### ISSUE TITLE
Fails to install on Pop!_OS 22.04

### ISSUE THREAD

**Issue by Bananas-Are-Yellow** [mkasberg/ghostty-ubuntu#1]
I'm using Pop!_OS 22.04 LTS, which is based on Ubuntu 22.04 LTS. I tried your latest deb file just now, but it failed for me. Here is the output:
```
$ sudo dpkg -i ghostty_*.deb
[sudo] password for david:
Selecting previously unselected package ghostty.
(Reading database ... 1038235 files and directories currently installed.)
Preparing to unpack ghostty_1.0.0-0.ppa1_amd64.deb ...
Unpacking ghostty (1.0.0-0~ppa1) ...
dpkg: dependency problems prevent configuration of ghostty:
 ghostty depends on libadwaita-1-0 (>= 1.5~beta); however:
  Version of libadwaita-1-0:amd64 on system is 1.1.7-0ubuntu0.22.04.1.
 ghostty depends on libc6 (>= 2.38); however:
  Version of libc6:amd64 on system is 2.35-0ubuntu3.8.
 ghostty depends on libglib2.0-0t64 (>= 2.42.0); however:
  Package libglib2.0-0t64 is not installed.

dpkg: error processing package ghostty (--install):
 dependency problems - leaving unconfigured
Processing triggers for mailcap (3.70+nmu1ubuntu1) ...
Processing triggers for gnome-menus (3.36.0-1ubuntu3) ...
Processing triggers for desktop-file-utils (0.26-1ubuntu3) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for man-db (2.10.2-1) ...
Errors were encountered while processing:
 ghostty
```

**Comment by mkasberg** [comment]
Thank you for the report! I built this on Ubuntu 24.10, I think some of the libraries are different. I will work on a version for 24.04 and 22.04.

**Comment by mkasberg** [comment]
@Bananas-Are-Yellow I've added a `.deb` for 22.04 to the releases page. Give that a shot?

https://github.com/mkasberg/ghostty-ubuntu/releases/tag/1.0.0-0-ppa1

**Comment by mkasberg** [comment]
Hmm, it's not obvious to me what the problem is there...

Can you see if `ghostty +crash-report` shows anything?

**Comment by Bananas-Are-Yellow** [comment]
```
$ ghostty +crash-report
No crash reports! 👻⏎
```

**Comment by Bananas-Are-Yellow** [comment]
Comparing my output with that shown [here](https://github.com/ghostty-org/ghostty/discussions/3537), it seems that all the messages are harmless, except for the last line that says "error: FileNotFound".

**Comment by Bananas-Are-Yellow** [comment]
I don't have time to investigate this evening, but it might be something to do with [this note](https://ghostty.org/docs/install/build):

> A recent GTK is required for Ghostty to work with Nvidia (GL) drivers under x11. Ubuntu 22.04 LTS has GTK 4.6 which is not new enough. Ubuntu 23.10 has GTK 4.12 and works. From [this discussion](https://discourse.gnome.org/t/opengl-context-version-not-respected-on-gtk4-rs/12162?u=cdehais) the problem was fixed in GTK by Dec 2022.

On my computer:
```
$ apt list libgtk-4-1
Listing... Done
libgtk-4-1/jammy-security,jammy-updates,now 4.6.9+ds-0ubuntu0.22.04.2 amd64 [installed,automatic]
libgtk-4-1/jammy-security,jammy-updates 4.6.9+ds-0ubuntu0.22.04.2 i386
```

**Comment by Bananas-Are-Yellow** [comment]
I'm now on my laptop, which is using Integrated graphics, not NVIDIA graphics. So that wasn't the reason.

**Comment by Bananas-Are-Yellow** [comment]
I built Ghostty from source using the steps below, and it runs fine on my laptop (Pop!_OS 22.04 LTS, integrated graphics).
```
sudo apt install libgtk-4-dev libadwaita-1-dev
# I was prompted to restart to continue installation

snap install zig --classic --beta
# installs Zig 0.13.0, which is the required version

zig build -p $HOME/.local -Doptimize=ReleaseFast
```

**Comment by Bananas-Are-Yellow** [comment]
> I don't have time to investigate this evening, but it might be something to do with [this note](https://ghostty.org/docs/install/build):
> 
> > A recent GTK is required for Ghostty to work with Nvidia (GL) drivers under x11. Ubuntu 22.04 LTS has GTK 4.6 which is not new enough. Ubuntu 23.10 has GTK 4.12 and works. From [this discussion](https://discourse.gnome.org/t/opengl-context-version-not-respected-on-gtk4-rs/12162?u=cdehais) the problem was fixed in GTK by Dec 2022.
> 
> On my computer:
> 
> ```
> $ apt list libgtk-4-1
> Listing... Done
> libgtk-4-1/jammy-security,jammy-updates,now 4.6.9+ds-0ubuntu0.22.04.2 amd64 [installed,automatic]
> libgtk-4-1/jammy-security,jammy-updates 4.6.9+ds-0ubuntu0.22.04.2 i386
> ```
> 
> I am indeed using Nvidia drivers under x11.

I built Ghostty from source on this desktop computer too, which uses NVIDIA drivers and X11, but it fails for the reasons described in that note.
```
$ ghostty
info: ghostty version=1.0.0-dev+0000000
info: ghostty build optimize=ReleaseFast
info: runtime=apprt.Runtime.gtk
info: font_backend=font.main.Backend.fontconfig_freetype
info: dependency harfbuzz=8.4.0
info: dependency fontconfig=21402
info: renderer=renderer.OpenGL
info: libxev backend=main.Backend.io_uring
info(os): setlocale from env result=en_US.UTF-8
info(gtk): GTK version build=4.6.9 runtime=4.6.9
info: optional config file not found, not loading path=/home/david/.config/ghostty/config
info(config): default shell source=env value=/usr/bin/fish
info(sentry): sentry envelope does not contain crash, discarding
"vulkan-disable" is only available when building GTK with G_ENABLE_DEBUG. See GDK_DEBUG=help
info(gtk): libadwaita version build=1.1.7 runtime=1.1.7
error(gtk): unable to get current color scheme: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such method “ReadOne”
info(grid): loaded OpenGL 3.2
warning(grid): OpenGL version is too old. Ghostty requires OpenGL 3.3
error(gtk_surface): surface failed to realize: error.OpenGLOutdated
fish: Job 1, 'ghostty' terminated by signal SIGSEGV (Address boundary error)
```
Nothing to do with your deb in this case.

**Comment by mkasberg** [comment]
I've been able to reproduce `error: FileNotFound` in a PopOS VM. Not sure whether it's due to my build or not but I'll investigate some more as I have time.

**Comment by mkasberg** [comment]
I have a working (built on the VM) and non-working (from my deb) Ghostty in PopOS 22.04. The only difference for me in the `ghostty +version` output is the working one has `- libX11     : enabled` and the other does not. I need to look into this further to understand the cause.

**Comment by Bananas-Are-Yellow** [comment]
> I have a working (built on the VM) and non-working (from my deb) Ghostty in PopOS 22.04. The only difference for me in the `ghostty +version` output is the working one has `- libX11 : enabled` and the other does not. I need to look into this further to understand the cause.

For reference, here is the `ghostty +version` output from the laptop where I built the source and Ghostty works.
```
$ ghostty +version
Ghostty 1.0.0-dev+0000000

Version
  - version: 1.0.0-dev+0000000
  - channel: tip
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.6.9
    runtime    : 4.6.9
  - libadwaita : enabled
    build      : 1.1.7
    runtime    : 1.1.7
```
I don't see any mention of libX11. The laptop uses X11, according to the About tab of my Settings app:

![image](https://github.com/user-attachments/assets/57ff574b-1df3-417f-834d-ce6bbc087113)

**Comment by maheshrijal** [comment]
I just wanted to pop in and mention that I'm on PopOS 2204 as well. I was able to run the app without issues. I'm running NVIDIA graphics.

**Comment by Suman-JS** [comment]
I was having the same issue, but  'ghostty_1.0.0-0.ppa1_amd64_22.04.deb' this worked without any issue

**Comment by crkarthik11** [comment]
I am also running pop os with below config

OS : Pop!_OS 22.04 LTS 
OS Type : 64 Bit
Gnome Version: 42.9
Windowing System: X11

```

ghostty +version
Ghostty 1.0.0-dev+0000000

Version
  - version: 1.0.0-dev+0000000
  - channel: tip
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.6.9
    runtime    : 4.6.9
  - libadwaita : enabled
    build      : 1.1.7
    runtime    : 1.1.7
```
After installing 

`sudo apt install libx11-dev`

I am getting below error

```
ghostty
info: ghostty version=1.0.0-dev+0000000
info: ghostty build optimize=ReleaseFast
info: runtime=apprt.Runtime.gtk
info: font_backend=font.main.Backend.fontconfig_freetype
info: dependency harfbuzz=2.7.4
info: dependency fontconfig=21301
info: renderer=renderer.OpenGL
info: libxev backend=main.Backend.io_uring
info(os): setlocale from env result=en_US.UTF-8
info(gtk): GTK version build=4.6.9 runtime=4.6.9
info: optional config file not found, not loading path=/home/delta/.config/ghostty/config
info(config): default shell source=env value=/bin/bash
"vulkan-disable" is only available when building GTK with G_ENABLE_DEBUG. See GDK_DEBUG=help
info(gtk): libadwaita version build=1.1.7 runtime=1.1.7
error(gtk): unable to get current color scheme: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such method “ReadOne”
info(grid): loaded OpenGL 3.2
warning(grid): OpenGL version is too old. Ghostty requires OpenGL 3.3
error(gtk_surface): surface failed to realize: error.OpenGLOutdated
```

But system OpenGL version is as below

```
$glxinfo | grep "OpenGL version"
OpenGL version string: 4.6.0 NVIDIA 560.35.03
```

**Comment by Bananas-Are-Yellow** [comment]
If you have NVIDIA drivers and are using X11, Ghostty won't work. As well as the note I mentioned above, there is more information [here](https://github.com/ghostty-org/ghostty/discussions/3159).

But this has nothing to do with the deb file. It won't work if you build the source yourself too. The issue with the deb file is when it should work, i.e. when you are not using NVIDIA drivers and X11.

If the deb works for some people, this suggests there is a missing dependency that some people already have present.

**Comment by uncomplexity** [comment]
got 1.0.0 installed on Ubuntu 22.04.5 LTS! looking good <3

noticed 1.0.1-0~ppa1 doesn't have it though, hope 22.x gets part of releases in the future (no rush no pressure), thank you

**Comment by mkasberg** [comment]
I removed builds for 22.04 because I didn't want to publish a build that would be broken for most people. But based on the number of people here saying it works for them, it's clear that's not the case (even if it's broken for some people). I think I'll put it back while I continue debugging the problem here. It's there now on the [Release](https://github.com/mkasberg/ghostty-ubuntu/releases/tag/1.0.1-0-ppa1) page.

I think @Bananas-Are-Yellow is on the right track with a missing dependency.

**Comment by mkasberg** [comment]
I've confirmed in a VM that `sudo apt install libx11-dev` seems to fix the problem on PopOS 22.04. I don't understand why though, and it seems wrong to declare a _runtime_ dependency on `libx11-dev`. So this might be a good workaround for you, but I don't necessarily think it's the right fix for the package...

**Comment by Bananas-Are-Yellow** [comment]
Just a thought. Some people would have been building Ghostty from source, as early adopters. It might be that the deb file works if you have previously built from source, since you would have installed `libgtk-4-dev` and `libadwaita-1-dev` and perhaps other things.

The deb file didn't work for me, but I had never built Ghostty from source before. I have done now.

I wonder what the steps are to completely remove a source-built Ghostty installation? I tried this, but it didn't seem to do anything at all:
```
zig build uninstall -p $HOME/.local
```

**Comment by mkasberg** [comment]
> I wonder what the steps are to completely remove a source-built Ghostty installation?

I had to do this a couple days ago (when I change my own install from source built to `.local` to a `.deb` package). I found that `zig build uninstall -p $HOME/.local` removed some things but didn't seem to get all of them. Here are the files that got installed (replace `zig-out` with `$HOME/.local`) (I found these just by listing a build I did to `zig-out`). You could just remove them. It's not as bad as it looks since a few `ghostty` directories contain many files.

```
$ find zig-out -type f
zig-out/bin/ghostty
zig-out/share/man/.placeholder
zig-out/share/applications/com.mitchellh.ghostty.desktop
zig-out/share/vim/vimfiles/syntax/ghostty.vim
zig-out/share/vim/vimfiles/ftdetect/ghostty.vim
zig-out/share/vim/vimfiles/ftplugin/ghostty.vim
zig-out/share/bat/syntaxes/ghostty.sublime-syntax
zig-out/share/nvim/site/syntax/ghostty.vim
zig-out/share/nvim/site/ftdetect/ghostty.vim
zig-out/share/nvim/site/ftplugin/ghostty.vim
zig-out/share/ghostty/shell-integration/elvish/lib/ghostty-integration.elv
zig-out/share/ghostty/shell-integration/fish/vendor_conf.d/ghostty-shell-integration.fish
zig-out/share/ghostty/shell-integration/bash/ghostty.bash
zig-out/share/ghostty/shell-integration/bash/bash-preexec.sh
zig-out/share/ghostty/shell-integration/zsh/.zshenv
zig-out/share/ghostty/shell-integration/zsh/ghostty-integration
zig-out/share/ghostty/themes/darkermatrix
zig-out/share/ghostty/themes/IC_Green_PPL
zig-out/share/ghostty/themes/Borland
zig-out/share/ghostty/themes/iTerm2 Tango Light
zig-out/share/ghostty/themes/Lab Fox
zig-out/share/ghostty/themes/Popping and Locking
zig-out/share/ghostty/themes/PencilDark
zig-out/share/ghostty/themes/NightLion v1
zig-out/share/ghostty/themes/coffee_theme
zig-out/share/ghostty/themes/Wryan
zig-out/share/ghostty/themes/FunForrest
zig-out/share/ghostty/themes/synthwave
zig-out/share/ghostty/themes/Atom
zig-out/share/ghostty/themes/jubi
zig-out/share/ghostty/themes/Smyck
zig-out/share/ghostty/themes/Chalkboard
zig-out/share/ghostty/themes/Firewatch
zig-out/share/ghostty/themes/GitHub Dark
zig-out/share/ghostty/themes/Abernathy
zig-out/share/ghostty/themes/Wombat
zig-out/share/ghostty/themes/SpaceGray Eighties Dull
zig-out/share/ghostty/themes/zenburned
zig-out/share/ghostty/themes/Highway
zig-out/share/ghostty/themes/MonaLisa
zig-out/share/ghostty/themes/Misterioso
zig-out/share/ghostty/themes/NightLion v2
zig-out/share/ghostty/themes/Chalk
zig-out/share/ghostty/themes/Ocean
zig-out/share/ghostty/themes/BlueBerryPie
zig-out/share/ghostty/themes/Overnight Slumber
zig-out/share/ghostty/themes/arcoiris
zig-out/share/ghostty/themes/BlueDolphin
zig-out/share/ghostty/themes/lovelace
zig-out/share/ghostty/themes/Oceanic-Next
zig-out/share/ghostty/themes/Jellybeans
zig-out/share/ghostty/themes/matrix
zig-out/share/ghostty/themes/Seafoam Pastel
zig-out/share/ghostty/themes/Builtin Dark
zig-out/share/ghostty/themes/iTerm2 Dark Background
zig-out/share/ghostty/themes/Mellifluous
zig-out/share/ghostty/themes/DjangoRebornAgain
zig-out/share/ghostty/themes/IR_Black
zig-out/share/ghostty/themes/Shaman
zig-out/share/ghostty/themes/Builtin Solarized Dark
zig-out/share/ghostty/themes/Hopscotch.256
zig-out/share/ghostty/themes/OceanicMaterial
zig-out/share/ghostty/themes/Builtin Tango Light
zig-out/share/ghostty/themes/xcodewwdc
zig-out/share/ghostty/themes/AlienBlood
zig-out/share/ghostty/themes/Sublette
zig-out/share/ghostty/themes/Peppermint
zig-out/share/ghostty/themes/Neutron
zig-out/share/ghostty/themes/Espresso Libre
zig-out/share/ghostty/themes/Violet Light
zig-out/share/ghostty/themes/Monokai Soda
zig-out/share/ghostty/themes/WildCherry
zig-out/share/ghostty/themes/SpaceGray Bright
zig-out/share/ghostty/themes/Grass
zig-out/share/ghostty/themes/DoomOne
zig-out/share/ghostty/themes/Cobalt Neon
zig-out/share/ghostty/themes/SoftServer
zig-out/share/ghostty/themes/UltraDark
zig-out/share/ghostty/themes/CLRS
zig-out/share/ghostty/themes/tokyonight-storm
zig-out/share/ghostty/themes/Wez
zig-out/share/ghostty/themes/Github
zig-out/share/ghostty/themes/Sakura
zig-out/share/ghostty/themes/flexoki-dark
zig-out/share/ghostty/themes/Firefly Traditional
zig-out/share/ghostty/themes/kanagawabones
zig-out/share/ghostty/themes/Material
zig-out/share/ghostty/themes/vesper
zig-out/share/ghostty/themes/DimmedMonokai
zig-out/share/ghostty/themes/DjangoSmooth
zig-out/share/ghostty/themes/HaX0R_GR33N
zig-out/share/ghostty/themes/CGA
zig-out/share/ghostty/themes/nightfox
zig-out/share/ghostty/themes/Raycast_Dark
zig-out/share/ghostty/themes/Mathias
zig-out/share/ghostty/themes/Twilight
zig-out/share/ghostty/themes/Tomorrow Night Bright
zig-out/share/ghostty/themes/Rippedcasts
zig-out/share/ghostty/themes/Builtin Tango Dark
zig-out/share/ghostty/themes/SynthwaveAlpha
zig-out/share/ghostty/themes/Dimidium
zig-out/share/ghostty/themes/Cobalt2
zig-out/share/ghostty/themes/RetroLegends
zig-out/share/ghostty/themes/Sundried
zig-out/share/ghostty/themes/Aardvark Blue
zig-out/share/ghostty/themes/Galizur
zig-out/share/ghostty/themes/cyberpunk
zig-out/share/ghostty/themes/catppuccin-latte
zig-out/share/ghostty/themes/Pnevma
zig-out/share/ghostty/themes/Novel
zig-out/share/ghostty/themes/Darkside
zig-out/share/ghostty/themes/Red Alert
zig-out/share/ghostty/themes/Dracula+
zig-out/share/ghostty/themes/Jackie Brown
zig-out/share/ghostty/themes/deep
zig-out/share/ghostty/themes/Solarized Darcula
zig-out/share/ghostty/themes/Dracula
zig-out/share/ghostty/themes/Unikitty
zig-out/share/ghostty/themes/seoulbones_dark
zig-out/share/ghostty/themes/Alabaster
zig-out/share/ghostty/themes/NvimDark
zig-out/share/ghostty/themes/Subliminal
zig-out/share/ghostty/themes/HaX0R_BLUE
zig-out/share/ghostty/themes/WarmNeon
zig-out/share/ghostty/themes/CutiePro
zig-out/share/ghostty/themes/Scarlet Protocol
zig-out/share/ghostty/themes/MaterialDesignColors
zig-out/share/ghostty/themes/Operator Mono Dark
zig-out/share/ghostty/themes/iceberg-dark
zig-out/share/ghostty/themes/Guezwhoz
zig-out/share/ghostty/themes/C64
zig-out/share/ghostty/themes/Builtin Solarized Light
zig-out/share/ghostty/themes/neobones_light
zig-out/share/ghostty/themes/Espresso
zig-out/share/ghostty/themes/BlulocoDark
zig-out/share/ghostty/themes/Elementary
zig-out/share/ghostty/themes/Hivacruz
zig-out/share/ghostty/themes/FishTank
zig-out/share/ghostty/themes/VibrantInk
zig-out/share/ghostty/themes/Hurtado
zig-out/share/ghostty/themes/Whimsy
zig-out/share/ghostty/themes/iceberg-light
zig-out/share/ghostty/themes/zenbones_dark
zig-out/share/ghostty/themes/Spiderman
zig-out/share/ghostty/themes/Vaughn
zig-out/share/ghostty/themes/niji
zig-out/share/ghostty/themes/Apple System Colors
zig-out/share/ghostty/themes/Red Sands
zig-out/share/ghostty/themes/flexoki-light
zig-out/share/ghostty/themes/Retro
zig-out/share/ghostty/themes/dayfox
zig-out/share/ghostty/themes/Blue Matrix
zig-out/share/ghostty/themes/Tinacious Design (Light)
zig-out/share/ghostty/themes/zenwritten_light
zig-out/share/ghostty/themes/LiquidCarbon
zig-out/share/ghostty/themes/SleepyHollow
zig-out/share/ghostty/themes/Everforest Dark - Hard
zig-out/share/ghostty/themes/Zenburn
zig-out/share/ghostty/themes/Relaxed
zig-out/share/ghostty/themes/Kibble
zig-out/share/ghostty/themes/xcodelight
zig-out/share/ghostty/themes/Monokai Vivid
zig-out/share/ghostty/themes/Adventure
zig-out/share/ghostty/themes/Builtin Pastel Dark
zig-out/share/ghostty/themes/DotGov
zig-out/share/ghostty/themes/Argonaut
zig-out/share/ghostty/themes/Molokai
zig-out/share/ghostty/themes/Hybrid
zig-out/share/ghostty/themes/farmhouse-light
zig-out/share/ghostty/themes/AdventureTime
zig-out/share/ghostty/themes/Earthsong
zig-out/share/ghostty/themes/OneHalfLight
zig-out/share/ghostty/themes/xcodedarkhc
zig-out/share/ghostty/themes/nord-light
zig-out/share/ghostty/themes/Spacedust
zig-out/share/ghostty/themes/OneHalfDark
zig-out/share/ghostty/themes/Purple Rain
zig-out/share/ghostty/themes/Ryuuko
zig-out/share/ghostty/themes/Oxocarbon
zig-out/share/ghostty/themes/The Hulk
zig-out/share/ghostty/themes/Piatto Light
zig-out/share/ghostty/themes/catppuccin-macchiato
zig-out/share/ghostty/themes/nord
zig-out/share/ghostty/themes/Symfonic
zig-out/share/ghostty/themes/AtomOneLight
zig-out/share/ghostty/themes/Cyberdyne
zig-out/share/ghostty/themes/PaulMillr
zig-out/share/ghostty/themes/Fairyfloss
zig-out/share/ghostty/themes/Medallion
zig-out/share/ghostty/themes/Tomorrow Night Eighties
zig-out/share/ghostty/themes/heeler
zig-out/share/ghostty/themes/iTerm2 Smoooooth
zig-out/share/ghostty/themes/MaterialOcean
zig-out/share/ghostty/themes/Galaxy
zig-out/share/ghostty/themes/Homebrew
zig-out/share/ghostty/themes/Aurora
zig-out/share/ghostty/themes/Doom Peacock
zig-out/share/ghostty/themes/Nocturnal Winter
zig-out/share/ghostty/themes/Pandora
zig-out/share/ghostty/themes/Monokai Remastered
zig-out/share/ghostty/themes/BlulocoLight
zig-out/share/ghostty/themes/Thayer Bright
zig-out/share/ghostty/themes/Night Owlish Light
zig-out/share/ghostty/themes/iTerm2 Solarized Dark
zig-out/share/ghostty/themes/idleToes
zig-out/share/ghostty/themes/Paraiso Dark
zig-out/share/ghostty/themes/darkmatrix
zig-out/share/ghostty/themes/shades-of-purple
zig-out/share/ghostty/themes/Tomorrow
zig-out/share/ghostty/themes/farmhouse-dark
zig-out/share/ghostty/themes/wilmersdorf
zig-out/share/ghostty/themes/Floraverse
zig-out/share/ghostty/themes/UnderTheSea
zig-out/share/ghostty/themes/3024 Day
zig-out/share/ghostty/themes/Lavandula
zig-out/share/ghostty/themes/gruber-darker
zig-out/share/ghostty/themes/Konsolas
zig-out/share/ghostty/themes/PaleNightHC
zig-out/share/ghostty/themes/Kanagawa Dragon
zig-out/share/ghostty/themes/rebecca
zig-out/share/ghostty/themes/synthwave-everything
zig-out/share/ghostty/themes/Builtin Light
zig-out/share/ghostty/themes/seoulbones_light
zig-out/share/ghostty/themes/iTerm2 Solarized Light
zig-out/share/ghostty/themes/Fideloper
zig-out/share/ghostty/themes/rose-pine-moon
zig-out/share/ghostty/themes/Ubuntu
zig-out/share/ghostty/themes/xcodelighthc
zig-out/share/ghostty/themes/zenbones_light
zig-out/share/ghostty/themes/Sugarplum
zig-out/share/ghostty/themes/Ollie
zig-out/share/ghostty/themes/Chester
zig-out/share/ghostty/themes/Django
zig-out/share/ghostty/themes/iTerm2 Pastel Dark Background
zig-out/share/ghostty/themes/Slate
zig-out/share/ghostty/themes/catppuccin-mocha
zig-out/share/ghostty/themes/FrontEndDelight
zig-out/share/ghostty/themes/Obsidian
zig-out/share/ghostty/themes/Grey-green
zig-out/share/ghostty/themes/CyberpunkScarletProtocol
zig-out/share/ghostty/themes/Snazzy
zig-out/share/ghostty/themes/ENCOM
zig-out/share/ghostty/themes/Tomorrow Night Blue
zig-out/share/ghostty/themes/MaterialDarker
zig-out/share/ghostty/themes/Belafonte Day
zig-out/share/ghostty/themes/Ciapre
zig-out/share/ghostty/themes/FirefoxDev
zig-out/share/ghostty/themes/Rouge 2
zig-out/share/ghostty/themes/iTerm2 Light Background
zig-out/share/ghostty/themes/Raycast_Light
zig-out/share/ghostty/themes/zenbones
zig-out/share/ghostty/themes/LiquidCarbonTransparentInverse
zig-out/share/ghostty/themes/Pro
zig-out/share/ghostty/themes/Treehouse
zig-out/share/ghostty/themes/Desert
zig-out/share/ghostty/themes/Harper
zig-out/share/ghostty/themes/Mirage
zig-out/share/ghostty/themes/tokyonight-day
zig-out/share/ghostty/themes/MaterialDark
zig-out/share/ghostty/themes/SpaceGray Eighties
zig-out/share/ghostty/themes/GruvboxDark
zig-out/share/ghostty/themes/Tomorrow Night
zig-out/share/ghostty/themes/duckbones
zig-out/share/ghostty/themes/Bright Lights
zig-out/share/ghostty/themes/Flatland
zig-out/share/ghostty/themes/Calamity
zig-out/share/ghostty/themes/Apple Classic
zig-out/share/ghostty/themes/tokyonight
zig-out/share/ghostty/themes/Banana Blueberry
zig-out/share/ghostty/themes/iTerm2 Tango Dark
zig-out/share/ghostty/themes/Glacier
zig-out/share/ghostty/themes/kurokula
zig-out/share/ghostty/themes/Pro Light
zig-out/share/ghostty/themes/Seti
zig-out/share/ghostty/themes/iTerm2 Default
zig-out/share/ghostty/themes/Kolorit
zig-out/share/ghostty/themes/Everblush
zig-out/share/ghostty/themes/AtelierSulphurpool
zig-out/share/ghostty/themes/ToyChest
zig-out/share/ghostty/themes/Square
zig-out/share/ghostty/themes/Tomorrow Night Burns
zig-out/share/ghostty/themes/Rapture
zig-out/share/ghostty/themes/Hopscotch
zig-out/share/ghostty/themes/Hacktober
zig-out/share/ghostty/themes/IC_Orange_PPL
zig-out/share/ghostty/themes/Duotone Dark
zig-out/share/ghostty/themes/Tinacious Design (Dark)
zig-out/share/ghostty/themes/embers-dark
zig-out/share/ghostty/themes/Fahrenheit
zig-out/share/ghostty/themes/Horizon
zig-out/share/ghostty/themes/UltraViolent
zig-out/share/ghostty/themes/rose-pine-dawn
zig-out/share/ghostty/themes/Elemental
zig-out/share/ghostty/themes/Kanagawa Wave
zig-out/share/ghostty/themes/rose-pine
zig-out/share/ghostty/themes/Dark Pastel
zig-out/share/ghostty/themes/3024 Night
zig-out/share/ghostty/themes/mellow
zig-out/share/ghostty/themes/GruvboxLight
zig-out/share/ghostty/themes/Royal
zig-out/share/ghostty/themes/Neon
zig-out/share/ghostty/themes/Breeze
zig-out/share/ghostty/themes/Violet Dark
zig-out/share/ghostty/themes/Andromeda
zig-out/share/ghostty/themes/Ayu Mirage
zig-out/share/ghostty/themes/NvimLight
zig-out/share/ghostty/themes/Broadcast
zig-out/share/ghostty/themes/Dark+
zig-out/share/ghostty/themes/Later This Evening
zig-out/share/ghostty/themes/Teerb
zig-out/share/ghostty/themes/catppuccin-frappe
zig-out/share/ghostty/themes/ChallengerDeep
zig-out/share/ghostty/themes/BirdsOfParadise
zig-out/share/ghostty/themes/Urple
zig-out/share/ghostty/themes/CrayonPonyFish
zig-out/share/ghostty/themes/LiquidCarbonTransparent
zig-out/share/ghostty/themes/HaX0R_R3D
zig-out/share/ghostty/themes/vimbones
zig-out/share/ghostty/themes/Mariana
zig-out/share/ghostty/themes/ForestBlue
zig-out/share/ghostty/themes/Belafonte Night
zig-out/share/ghostty/themes/Afterglow
zig-out/share/ghostty/themes/Dark Modern
zig-out/share/ghostty/themes/Framer
zig-out/share/ghostty/themes/Man Page
zig-out/share/ghostty/themes/ayu
zig-out/share/ghostty/themes/PencilLight
zig-out/share/ghostty/themes/Flat
zig-out/share/ghostty/themes/Batman
zig-out/share/ghostty/themes/GruvboxDarkHard
zig-out/share/ghostty/themes/midnight-in-mojave
zig-out/share/ghostty/themes/Grape
zig-out/share/ghostty/themes/JetBrains Darcula
zig-out/share/ghostty/themes/Terminal Basic
zig-out/share/ghostty/themes/Spring
zig-out/share/ghostty/themes/Japanesque
zig-out/share/ghostty/themes/terafox
zig-out/share/ghostty/themes/gruvbox-material
zig-out/share/ghostty/themes/primary
zig-out/share/ghostty/themes/Laser
zig-out/share/ghostty/themes/purplepeter
zig-out/share/ghostty/themes/SeaShells
zig-out/share/ghostty/themes/Hipster Green
zig-out/share/ghostty/themes/Solarized Dark Higher Contrast
zig-out/share/ghostty/themes/idea
zig-out/share/ghostty/themes/Neopolitan
zig-out/share/ghostty/themes/ayu_light
zig-out/share/ghostty/themes/Blazer
zig-out/share/ghostty/themes/zenwritten_dark
zig-out/share/ghostty/themes/Tango Half Adapted
zig-out/share/ghostty/themes/Tango Adapted
zig-out/share/ghostty/themes/Arthur
zig-out/share/ghostty/themes/SpaceGray
zig-out/share/ghostty/themes/neobones_dark
zig-out/share/ghostty/themes/xcodedark
zig-out/share/ghostty/themes/Brogrammer
zig-out/share/ghostty/themes/N0tch2k
zig-out/share/ghostty/themes/Red Planet
zig-out/share/ghostty/themes/Solarized Dark - Patched
zig-out/share/ghostty/themes/Hardcore
zig-out/share/fish/vendor_completions.d/ghostty.fish
zig-out/share/bash-completion/completions/ghostty.bash
zig-out/share/zsh/site-functions/_ghostty
zig-out/share/terminfo/ghostty.terminfo
zig-out/share/terminfo/x/xterm-ghostty
zig-out/share/terminfo/ghostty.termcap
zig-out/share/icons/hicolor/16x16@2/apps/com.mitchellh.ghostty.png
zig-out/share/icons/hicolor/256x256@2/apps/com.mitchellh.ghostty.png
zig-out/share/icons/hicolor/32x32/apps/com.mitchellh.ghostty.png
zig-out/share/icons/hicolor/128x128/apps/com.mitchellh.ghostty.png
zig-out/share/icons/hicolor/32x32@2/apps/com.mitchellh.ghostty.png
zig-out/share/icons/hicolor/16x16/apps/com.mitchellh.ghostty.png
zig-out/share/icons/hicolor/128x128@2/apps/com.mitchellh.ghostty.png
zig-out/share/icons/hicolor/256x256/apps/com.mitchellh.ghostty.png
zig-out/share/icons/hicolor/512x512/apps/com.mitchellh.ghostty.png
zig-out/share/kio/servicemenus/com.mitchellh.ghostty.desktop
```

**Comment by Bananas-Are-Yellow** [comment]
Thanks. The following commands seem to cover everything:
```
cd $HOME/.local
rm -rf share/ghostty
rm bin/ghostty
find share -type f -name '*ghostty*' -delete
find share -type l -name '*ghostty*' -delete
```
I assume it's harmless to leave the odd empty parent directory around.

By the way, I didn't have `share/man/.placeholder` in my `$HOME/.local` directory.

**Comment by Bananas-Are-Yellow** [comment]
I removed the source build using the commands I listed above, then I reversed the other steps I had taken in the process of building the source:
```
sudo apt remove libgtk-4-dev libadwaita-1-dev
snap remove zig
```
Then I installed your latest 22.04 deb file, and it works. But I see you updated the deb file a few hours ago. Does it now contain the `libx11-dev` workaround?

**Comment by konsumer** [comment]
I am also on Pop!_OS 22.04 LTS, and installed the latest 22.04 deb. Installation when ok, but I get same `OpenGL version is too old. Ghostty requires OpenGL 3.3`

```
❯ /usr/bin/ghostty
info: ghostty version=1.0.1
info: ghostty build optimize=ReleaseFast
info: runtime=apprt.Runtime.gtk
info: font_backend=font.main.Backend.fontconfig_freetype
info: dependency harfbuzz=2.7.4
info: dependency fontconfig=21301
info: renderer=renderer.OpenGL
info: libxev backend=main.Backend.io_uring
info(os): setlocale from env result=en_US.UTF-8
info(gtk): GTK version build=4.6.9 runtime=4.6.9
info(config): creating template config file: path=/home/konsumer/.config/ghostty/config
warning(config): error creating template config file err=error.FileNotFound
info(config): default shell source=env value=/usr/bin/zsh
"vulkan-disable" is only available when building GTK with G_ENABLE_DEBUG. See GDK_DEBUG=help
warning(sentry): crash report written to disk path=/home/konsumer/.local/state/ghostty/crash/6e2eae62-22ef-4dd0-f643-749926a6b710.ghosttycrash
info(gtk): libadwaita version build=1.1.7 runtime=1.1.7
info(grid): loaded OpenGL 3.2
warning(grid): OpenGL version is too old. Ghostty requires OpenGL 3.3
error(gtk_surface): surface failed to realize: error.OpenGLOutdated
[1]    12017 segmentation fault (core dumped)  /usr/bin/ghostty
```

**Comment by mkasberg** [comment]
> Then I installed your latest 22.04 deb file, and it works. But I see you updated the deb file a few hours ago. Does it now contain the libx11-dev workaround?

No, I did not (yet?) delcare this as a dependency.

**Comment by Bananas-Are-Yellow** [comment]
> > Then I installed your latest 22.04 deb file, and it works. But I see you updated the deb file a few hours ago. Does it now contain the libx11-dev workaround?
> 
> No, I did not (yet?) delcare this as a dependency.

Then what changes did you make to the 22.04 deb today? I think I've undone everything I did when I built from source, and now your deb works on my laptop when it didn't before. Did you do something to make it work, or have I forgotten some change I may have made when building from source?

Does the latest 22.04 deb work for you in your Pop!_OS 22.04 WM, or do you still have the FileNotFound problem?

**Comment by konsumer** [comment]
Personally, I decided to update my distro. I like Pop (and 22.04 is most recent) but it sounds like 22.04 is just a bit old. Maybe Ubuntu-Budgie 24.10 will work better.

**Comment by Bananas-Are-Yellow** [comment]
> Personally, I decided to update my distro. I like Pop (and 22.04 is most recent) but it sounds like 22.04 is just a bit old. Maybe Ubuntu-Budgie 24.10 will work better.

I think most people on Pop!_OS, including me, are waiting patiently for the upcoming COSMIC release, which is now at Alpha 4 stage. Hopefully not long to wait now. The alpha is on 24.04 LTS, but who knows, maybe it will be 25.04 by the time it is finally released.

**Comment by choyrim** [comment]
on my ubuntu 22.04, i was also able to install ghostty_1.0.0-0.ppa1_amd64_22.04.deb but not ghostty_1.0.1-0.ppa2_amd64_22.04.deb. 1.0.1-ppa2 failed due to dependency on libglib2.0-0t64 (as in the original post).

both packages had a dependency on libonig5 which was easy to install.

**Comment by HadrienPatte** [comment]
> on my ubuntu 22.04, i was also able to install ghostty_1.0.0-0.ppa1_amd64_22.04.deb but not ghostty_1.0.1-0.ppa2_amd64_22.04.deb. 1.0.1-ppa2 failed due to dependency on libglib2.0-0t64 (as in the original post).

The `libglib2.0-0` package only got renamed to `libglib2.0-0t64` with the release of ubuntu 24.04, previous ubuntu releases (including ubuntu 22.04) had it as `libglib2.0-0`.
See:
* [`libglib2.0-0`](https://packages.ubuntu.com/jammy/libglib2.0-0) on 22.04
* [`libglib2.0-0t64`](https://packages.ubuntu.com/noble/libglib2.0-0t64) on 24.04

**Comment by dblume** [comment]
Came here to say exactly the same thing as @HadrienPatte. I'm on ubuntu 22.04, and can install the ppa1 package, but not the ppa2 one, because the latter requires the t64 flavor of libglib2.0.0.

**Comment by dflock** [comment]
Same, getting `ghostty : Depends: libglib2.0-0t64 but it is not installable` with `ghostty_1.0.1-0.ppa2_amd64_22.04.deb` on Ubuntu 22.04:

```
OS: Xubuntu 22.04.5 LTS x86_64
Kernel: 6.8.0-50-generic
Uptime: 6 days, 17 hours, 27 mins
Shell: bash 5.2.37
Resolution: 1920x1080, 1920x1080, 1920x1080
DE: Xfce 4.16
WM: Xfwm4
WM Theme: Flat-Remix-Dark-XFWM
Theme: Flat-Remix-GTK-Blue-Darkest-Solid [GTK2], Arc-Dark [GTK3]
Icons: Flat-Remix-Blue-Dark [GTK2], Papirus-Dark [GTK3]
Terminal: ghostty
CPU: Intel i7-6700K (8) @ 4.200GHz
GPU: Intel HD Graphics 530
Memory: 20325MiB / 64185MiB
```

**Comment by mkasberg** [comment]
Thanks for the reports! I will look into this to see what changed. Knowing to look for [libglib2.0-0t64](https://packages.ubuntu.com/noble/libglib2.0-0t64) is helpful!

**Comment by mkasberg** [comment]
I know what changed. You can expect a fix within the next day or so.

**Comment by mkasberg** [comment]
The libglib2.0 issue is fixed by version 1.0.1-0~ppa3.

**Comment by Tiryoh** [comment]
@mkasberg 1.0.1-0~ppa3 worked on my Ubuntu 22.04 laptop (without NVIDIA GPU). Thanks for the update!

**Comment by dflock** [comment]
Yep, wfm now on Xubuntu 22.04.5, thanks!

**Comment by mkasberg** [comment]
Closing this issue as it's pretty stale and I believe the problems have been fixed. Please open a new issue if there's a specific problem that remains.
